### PR TITLE
Include parent head in `CandidateReceipt`

### PR DESCRIPTION
**Pull request by rphmeier**
The relay chain only accepts `CandidateReceipt`s in `set_heads` who have the correct `parent_head`, and validators will only accept and validate `CandidateReceipt`s which do.

### CODE REVIEW

**Comment by burdges** [comment]
We talked about being able to advance this when the parachain misses a block, which presumably the parachain validators do, not collators.

File: runtime/common/src/parachains.rs [hunk 1]
```diff
@@ -781,6 +783,14 @@ impl<T: Trait> Module<T> {
 			let validator_group = validator_groups.group_for(para_id)
 				.ok_or(Error::<T>::NoValidatorGroup)?;
 
+			let actual_head = Self::parachain_head(&para_id)
+				.map(primitives::parachain::HeadData);
+
+			ensure!(
+				actual_head.as_ref() == Some(&candidate.candidate.parent_head),
```
Comment (bkchr) [inline runtime/common/src/parachains.rs]: Do we really want to skip the inclusion of all parachain heads, when one is not correct?

(Will the validator that gave us this be slashed later?)

(I see that we do the same below for other checks as well, but the same question still holds)

Comment (rphmeier) [reply runtime/common/src/parachains.rs]: Yes - this gives us the property that every included head in the parachain block passes checks. These checks are relatively cheap, so it's fine to do them.

**Comment by rphmeier** [comment]
The reason to include the whole thing and not just a hash is that it makes it easier for fishermen - A fisherman presumably has access to a `Candidate` attestation, which will include the parent head data. Along with the `PoVBlock`, this is all of the information necessary to validate/fish. If it was only a hash, the fisherman would probably have to go into the old block's state in order to get the parent head data, which makes fishing less effective.

**Comment by burdges** [comment]
We do have a hash referring the parent `CandidateReceipt` too though, right?  If only because the relay chain cannot read the parachain's block header.

**Comment by rphmeier** [comment]
> If only because the relay chain cannot read the parachain's block header.

The `head_data` might contain a hash reference to some or all of the parent `head_data`. But yes, we cannot inspect it in the Relay chain state.

**Pull request merged**